packages/@aws-cdk/aws-cloudwatch/lib/metric.ts

```diff
@@ -610,8 +621,10 @@ export interface CreateAlarmOptions {
 
   /**
    * The value against which the specified statistic is compared.
+   *
+   * @default No threshold
    */
-  readonly threshold: number;
+  readonly threshold?: number;
```

Comment:
I realize that anomaly detection based alarms does not allow to set `threshold` and requires setting  `thresholdMetricId ` instead. 

However, making this property optional sacrifice the common use case ergonomics, in favor of the anomaly detection use case. Making this property optional means that we will need to add a synth time check that verifies:  _If this is  an anomaly detection based alarm and threshold is set then throw an error_ . While it can be done, it makes the developer experience not so great. 

Since there are several such subtleties in the anomaly detection alarm (e.g the allowed operators, the annotation),  it probably makes more sense to implement a `createAnomalyDetectionAlarm`.

I agree that this feature seemed trivial initially but it has many pitfalls, as such it will require more design work. Before we do another revision lets discuss the API in details, feel free to comment on this PR or create a new issue